index.js

```diff
@@ -119,19 +119,36 @@ function processRequestResponse (options, error, response, body) {
   var isChallengePresent;
   var isRedirectChallengePresent;
 
+  // Encoding is null so body should be a buffer object
   if (error || !body || !body.toString) {
     // Pure request error (bad connection, wrong url, etc)
     return callback(new errors.RequestError(error, options, response));
   }
 
-  stringBody = body.toString('utf8');
+  response.isCloudflare = response.statusCode > 499 &&
+    /^cloudflare/i.test('' + response.caseless.get('server')) &&
+    /text\/html/i.test('' + response.caseless.get('content-type'));
 
-  try {
-    validate(options, response, stringBody);
-  } catch (error) {
-    return callback(error);
+  if (response.isCloudflare) {
+    if (body.length < 1) {
+      // This is a 5xx Cloudflare response with an empty body.
+      return callback(new errors.CloudflareError(response.statusCode, options, response));
+    }
+
+    stringBody = body.toString('utf8');
+
+    try {
+      validate(options, response, stringBody);
+    } catch (error) {
+      return callback(error);
+    }
+  }
+
+  if (!response.isCloudflare || response.statusCode !== 503) {
```

Comment:
Are we sure that cloudflare always returns its pages with 503 code?

Reply:
Yup, and this is exactly what cloudflare-scrape has always checked for. No issues there.

Reply:
Correction, if it's the response contains a challenge then yes.

Reply:
@codemanki, Another option is to do a DNS lookup to identify cloudflare responses. Their CIDR network addresses are publicly noted at https://www.cloudflare.com/ips/ and they keep them updated in text form: https://www.cloudflare.com/ips-v4, https://www.cloudflare.com/ips-v6 . I don't think it's necessary though and wouldn't identify whether or not those responses contain challenges.

Reply:
@codemanki, Actually it isn't what they've always done. They introduced this back in July of 2016 in this commit: https://github.com/Anorov/cloudflare-scrape/commit/948470e2d97c10295cf8801e569aeaf78db09ea0

The comment that was there was to check to see if Anti-Bot was turned on and thats what's being done here. Rather than assuming every response is from cloudflare and checking for challenges. It determines whether or not the response if from cloudflare then only if anti-bot is turned on, does it check for challenges.

Reply:
@pro-src got it, thanks for the clarification!

Reply:
@codemanki You're always welcome!